src/app/users/associateServices.js

```diff
@@ -4,7 +4,7 @@ const {getAllServicesForUserInOrg} = require('./utils');
 
 const getAllAvailableServices = async (req) => {
   const allServices = await getAllServices(req.id);
-  let externalServices = allServices.services.filter(x => x.isExternalService === true);
+  let externalServices = allServices.services.filter(x => x.isExternalService === true && x.relyingParty && x.relyingParty.params && x.relyingParty.params.hideApprover !== 'true');
```

Comment:
I do not believe this will do what you intend. I believe you are trying to acheive "The service is an external service and it is not hidden from approvers" If that is the case then something like the following is probably what you want

`let externalServices = allServices.services.filter(x => x.isExternalService === true && !(x.relyingParty && x.relyingParty.params && x.relyingParty.params.hideApprover === 'true'));`

I would write some tests to cover:

1. External and not params
2. External, has params but no hideApprover
3. External, has params and hideApprover that equals false
4. External, has params and hideApprover that equals true
5. Not external, has params and hideApprover that equals false